sensor/openvas.go

```diff
@@ -0,0 +1,170 @@
+// OpenVAS component of the sensor. This module is responsible fot everything regarding OpenVAS
+package sensor
+
+import (
+	"errors"
+	"fmt"
+	"log"
+	"os"
+	"os/exec"
+	"strings"
+	"sync"
+)
+
+// rocessList represents a list of processes. It is used to manage processes
+// within different go routines.
+type processList struct {
+	procs map[string]*os.Process
+	mutex *sync.Mutex
+}
+
+// addProcess adds a Process to the Process list
+func (pl processList) addProcess(scan string, p *os.Process) error {
+	pl.mutex.Lock()
+	defer pl.mutex.Unlock()
+	if _, ok := pl.procs[scan]; ok {
+		return errors.New("process already exist")
+	}
+	pl.procs[scan] = p
+	return nil
+}
+
+// removeProcess removes a Process from the Process list
+func (pl processList) removeProcess(scan string) error {
+	pl.mutex.Lock()
+	defer pl.mutex.Unlock()
+	if _, ok := pl.procs[scan]; !ok {
+		return errors.New("process does not exist")
+	}
+	delete(pl.procs, scan)
+	return nil
+}
+
+var processes processList
+
+// StartScan starts scan with given scan-ID and process priority (-20 to 19,
+// lower is more prioritized)
+func StartScan(scan string, niceness int, sudo bool) error {
+	cmdString := make([]string, 0)
+
+	if niceness != 0 {
```

Comment:
shouldn't this be behind the line L58 ( `cmdString = append(cmdString, "openvas", "--scan-start", scan)` )?